### PR TITLE
Milestone2-Bookstore app

### PR DESCRIPTION
**Pull request by saimaAnis21**
In this project we did:
1- Prepare the directory structure
2- Implement the "App" component
3- Implement the BooksList component
4- Implement the Book component
5- Implement the BooksForm component
6- Define the Redux actions and reducers
7- Initialize the Redux store

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import BooksForm from '../containers/BooksForm';
+import BooksList from '../containers/BooksList';
+
+function App() {
+  return (
+    <div>
+      <BooksForm />
+      <BooksList />
+
```
Comment (bellom) [inline src/components/App.js]: Kiindly clear this empty white space.

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      <input />
+      <select>
+        {categories.map((cat) => (
+          <option key={cat} value={cat}>{cat}</option>
+        ))}
+        ;
+      </select>
+      <button type="submit"> Submit</button>
+
+    </div>
```
Comment (bellom) [inline src/containers/BooksForm.js]: According to the instructions on pathwright and getting data from user implementation, you need to use the form tag.
image

File: src/reducers/books.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const initialState = {
```
Comment (bellom) [inline src/reducers/books.js:1]: According to the instructions on pathwright, I think you need to move the `initialState` into `src/index.js`

image

[Pathwright](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376028/)

File: src/reducers/books.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  {
+    id: Math.random,
+    title: 'Second Book',
+    cat: 'Biography',
+  },
+  {
+    id: Math.random,
+    title: 'Third Book',
+    cat: 'History',
```
Comment (bellom) [inline src/reducers/books.js]: ### [Optional]
Kindly be aware that you can use uuidv4 to automatically generate ids. read more 👉 [npm uuid](https://www.npmjs.com/package/uuidv4)

Example:
```
import uuid from 'uuidv4'
id: uuid(): {title: 'First Book', category: 'Actioin'}
```

File: src/containers/BooksList.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+const BookList = (props) => {
+  const { books } = props;
+  return (
+    <table>
+      <th>Book ID</th>
+      <th>Title</th>
+      <th>Category</th>
+      { books.map((abook) => (<Book book={abook} key={abook} />))}
+
```
Comment (bellom) [inline src/containers/BooksList.js]: Kindly remove the white empty space which will make your code looks more professional and readable

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

**Comment by mimipeshy** [comment]
## STATUS: CHANGES REQUESTED 🔴 ♻️ 
Hello @saimaAnis21 and @Byabasaija 👋🏾 👋🏾 congratulations on making it this far 😄 . 

#### Highlights

- You have worked on some of the feedback given by the previous TSE ✔️ 

#### To improve
The following are yet to be implemented ⬇️ 
- [x] When I start your app with npm I get the following errors in the console, kindly fix them 

image
image

- [x] The previous TSE asked that you move initialState to src/index.js. Kindly refer to the previous comment they left and work on it too 😄 

That's all 😄, kindly submit another review when done
If you need any clarification, kindly post your comment here and tag me @mimipeshy and will be glad to help

Regards,
Peris Ndanu

File: src/reducers/books.js (hunk 3 above)
Comment (mimipeshy) [reply src/reducers/books.js:1]: - [ ] Kindly work on this comment as well

**Review by afanimartin — APPROVED** [review]
## Approved 👍🏿 👍🏿 👍🏿 
Changes made as requested. Merge and proceed to the next milestone.

### Martin Afani
#### Slack: whiz25

**Pull request merged**